earth_enterprise/src/common/CacheSizeCalculations.h

```diff
@@ -19,29 +19,44 @@
 #include "khTypes.h"
 #include <memory>
 #include <vector>
+#include <set>
+#include "notify.h"
+#include <qstring.h>
 
 // determine amount of memory used by generic objects
 template<class T>
 inline uint64 GetObjectSize(const T &obj) {
     return sizeof(obj);
 }
-// determine amount of memory used by a vector and all of it's contents
+// determine amount of memory used by a vector's contents
```

Comment:
It just occurred to me that you could have a generic method to get the size of objects in any container, which would work for vectors, sets, lists, etc. I think the only thing it wouldn't work for is maps. Something like:
```
template<class Container>
inline uint64 GetObjectSize(const Container &container) {
  uint64 total = 0;
  for (const auto & elem : container) {
    total += GetObjectSize(elem);
  }
  return total;
}
```

I guess you  might run into trouble where the compiler tried to use that for everything, though, so maybe it wouldn't work.

Reply:
The same issue here with the dependency problem that I was unsure of how to get around. I had definitely considered multiple options for addressing this, but this seemed like the simplest and most straightforward.

Reply:
If I include these other classes, I can add specific GetObjectSize methods that just call their respective GetSize method.